### PR TITLE
Plane: add Emergency stop for all motors

### PR DESCRIPTION
**Pull request by IamPete1**
This adds aux switch e stop to quadplanes 

When switched all  'Qmotors' will stop. If disarmed and estop is triggered the vehicle will refuse to arm in any-mode (so long as system arming checks are on). If armed in a none qmode and estop is triggerd then vehicle will refuse to switch into a Qmode. None Qmodes should work as normal. 

Tested in SITL, motor estop is aux function 31. The only thing (i think) missing is to log a dataflash message when triggered, copter has a nice system for this that is not set up in plane, so i have left it for the time being.

### CODE REVIEW

File: ArduPlane/RC_Channel.cpp [hunk 1]
```diff
@@ -32,6 +32,7 @@ void RC_Channel_Plane::init_aux_function(const RC_Channel::aux_func_t ch_option,
 {
     switch(ch_option) {
     // the following functions do not need to be initialised:
+    case MOTOR_ESTOP:
```
Comment (tridge) [inline ArduPlane/RC_Channel.cpp]: I think this means that if the switch starts in high position then motors won't initialised stopped?
I think estop should be enabled if the switch starts high

File: ArduPlane/ArduPlane.cpp [hunk 2]
```diff
@@ -945,4 +945,11 @@ void Plane::publish_osd_info()
 }
 #endif
 
+void Plane::set_motor_emergency_stop(bool b)
+{
+    if(motor_emergency_stop != b) {
```
Comment (peterbarker) [inline ArduPlane/ArduPlane.cpp]: Just do the assignment?

Comment (peterbarker) [reply ArduPlane/ArduPlane.cpp]: ... also, space after the `if`

File: ArduPlane/RC_Channel.cpp (hunk 1 above)
Comment (peterbarker) [reply ArduPlane/RC_Channel.cpp]: we should ensure the behaviour is the same as in Copter.  At some stage we should have this code common amongst vehicles.  Fewer places to make mistakes in a safty-critical part of the code.

It's not noted in here but @WickedShell pointed out on the devcall that the servo library already kinda-sorta knows which outputs have motors on them.

All our other vehicles could do with the concept of an estop.

File: ArduPlane/ArduPlane.cpp (hunk 2 above)
Comment (IamPete1) [reply ArduPlane/ArduPlane.cpp]: Your right, I was looking at this when I was doing the re-base. I just coped the function direct from copter, I thought there must have been some reason to do it this way.

File: ArduPlane/RC_Channel.cpp (hunk 1 above)
Comment (IamPete1) [reply ArduPlane/RC_Channel.cpp]: I will see what I can do

**Comment by IamPete1** [comment]
I have made a start on a universal E-stop https://github.com/ArduPilot/ardupilot/pull/10290 to supersede this.

**Comment by IamPete1** [comment]
Universal E-stop has now been merged so closing this

**Pull request closed without merging**